lib/prometheus.sh

```diff
@@ -42,6 +44,8 @@ deploy_prometheus_operator() {
 		mv kube-prometheus/manifests/prometheus-prometheus.yaml.tmp \
 			kube-prometheus/manifests/prometheus-prometheus.yaml
 
+		_setup_dashboard
```

Comment:
this should be run after a checking flag to avoid impact CI.

Reply:
wait, why the grafana dashboard been set up before grafana been deployed on the cluster?!

Reply:
@rootfs , I would like request to rollback or further review the changes for this PR.

Reply:
This just sets up the grafana configmap (please see the function below) - I can rename it to setup_grafana configmap and wrap it in the grafana flag

Reply:
> @rootfs , I would like request to rollback or further review the changes for this PR. With reasons:
> 
> 1. this the set dashboard always runs without grafana dashboard flag.

This was just the dashboard yaml setup - I can move it to the grafana wrapper - but editing the YAML should have no impact in the situation grafana isn't deployed. 

> 2. yq with the new command invoked, and seems the yq may moved to verify.sh or change a name to better understand, but not just run_yq as not a meaningful name.

This is editing the yaml files in place to add the dashboard and isn't a verification step

> 3. the hint and usage in main.sh is not updated.(nit)

I'm not sure what hint you are referring to. I'm using the same flags as set out by the project - can you please clarify?

> 4. it seems the impl for run_yq has error as file missing ref https://github.com/sustainable-computing-io/local-dev-cluster/actions/runs/8859334223/job/24328998652?pr=78#step:3:1055

I tried to see what file is missing, but I don't see it :( _run_yq runs before the `_load_prometheus_operator_images_to_local_registry` which looks like is what runs. at 1055 all I see is grafana deploy step ... :( I will look into this further

Reply:
Could push a new PR with something like this: https://github.com/sustainable-computing-io/local-dev-cluster/commit/a6f93da9cde05802245a265d3dca0cbf51a2f653